glean-core/rlb/src/dispatcher/global.rs

```diff
@@ -63,17 +63,27 @@ pub fn flush_init() -> Result<(), DispatchError> {
 ///
 /// This will initiate a shutdown of the worker thread
 /// and no new tasks will be processed after this.
-/// It will not block on the worker thread.
-pub fn try_shutdown() -> Result<(), DispatchError> {
-    guard().shutdown()
+pub fn shutdown() -> Result<(), DispatchError> {
+    guard().shutdown()?;
+
+    // After we issue the shutdown command, make sure to wait for the
+    // worker thread to join.
+    let mut lock = GLOBAL_DISPATCHER.write().unwrap();
+    let dispatcher = lock.as_mut().expect("Global dispatcher has gone missing");
+
+    if let Some(worker) = dispatcher.worker.take() {
+        return worker.join().map_err(|_| DispatchError::WorkerPanic);
+    }
+
+    Ok(())
 }
 
 /// TEST ONLY FUNCTION.
 /// Resets the Glean state and triggers init again.
 pub(crate) fn reset_dispatcher() {
     // We don't care about shutdown errors, since they will
     // definitely happen if this
-    let _ = try_shutdown();
+    let _ = shutdown();
```

Comment:
I can't comment on the lines below (damn you, github!), but line 92 (comment) is now kinda wrong and the code could be simplified I think.
As the worker thread is already `join`ed on, it's correctly shut down. I _think_ we don't even need the `mem::replace` anymore and a simple `*lock = new_dispatcher` should work now instead.